webapp/src/main/java/com/box/l10n/mojito/service/scheduledjob/jobs/ScheduledThirdPartySync.java

```diff
@@ -0,0 +1,146 @@
+    ThirdPartySyncJobInput thirdPartySyncJobInput =
+        new ThirdPartySyncJobInput(scheduledJob, scheduledJobProperties);
+
+    try {
+      PollableFuture<Void> task =
+          quartzPollableTaskScheduler.scheduleJobWithCustomTimeout(
+              ThirdPartySyncJob.class, thirdPartySyncJobInput, "thirdPartySync", 3600L);
+      pollableTaskId = task.getPollableTask().getId();
+      // Wait for sync to complete
+      task.get();
+    } catch (Exception e) {
+      throw new JobExecutionException(e);
+    }
+  }
+
+  @Override
+  public void onSuccess(JobExecutionContext context) {
+    logger.info(
+        "Third-Party Sync succeeded for repository {}.", scheduledJob.getRepository().getName());
+
+    // Resolve PD incident
+    pagerDutyIntegrationService
+        .getDefaultPagerDutyClient()
+        .ifPresent(
+            pd -> {
+              try {
+                pd.resolveIncident(scheduledJob.getId());
+              } catch (PagerDutyException e) {
+                logger.error(
+                    "Couldn't send resolve PagerDuty notification for successful third party sync of repository: '{}' -",
+                    scheduledJob.getRepository().getName(),
+                    e);
+              }
+            });
+  }
+
+  @Override
+  public void onFailure(JobExecutionContext context, JobExecutionException jobException) {
+    logger.error(
+        "Third-Party Sync for repository {} has failed. Pollable Task ID: {}",
+        scheduledJob.getRepository().getName(),
+        pollableTaskId);
+
+    // Trigger PD incident
+    pagerDutyIntegrationService
+        .getDefaultPagerDutyClient()
+        .ifPresent(
+            pd -> {
+              String pollableTaskUrl =
+                  UriComponentsBuilder.fromHttpUrl(serverConfig.getUrl())
+                      .path("api/pollableTasks/" + pollableTaskId.toString())
+                      .build()
+                      .toUriString();
+
+              String scheduledJobUrl =
+                  UriComponentsBuilder.fromHttpUrl(serverConfig.getUrl())
+                      .path("api/jobs/" + scheduledJob.getId())
+                      .build()
+                      .toUriString();
+
+              String title =
+                  StrSubstitutor.replace(
+                      notificationTitle,
+                      ImmutableMap.of("repository", scheduledJob.getRepository().getName()),
+                      "{",
+                      "}");
+
+              PagerDutyPayload payload =
+                  new PagerDutyPayload(
+                      title,
+                      serverConfig.getUrl(),
+                      PagerDutyPayload.Severity.CRITICAL,
+                      ImmutableMap.of(
+                          "Pollable Task", pollableTaskUrl, "Scheduled Job", scheduledJobUrl));
+
+              try {
+                pd.triggerIncident(scheduledJob.getId(), payload);
+              } catch (PagerDutyException e) {
```

Comment:
Could we also emit a native metric here (e.g.:`scheduledJobFailedTriggeringPagerDutyIncident` or something shorter/nicer) for when we failed to open a PD incident though we should have succeeded? (See: `meterRegistry.counter`)

We can then setup an alert based on that metric to alert us of that issue (missing real incidents).

Reply:
This is a really nice idea, will look into this now!

Reply:
Looks good to me!